VerilogForVerificationStyle.md

```diff
@@ -0,0 +1,850 @@
+# lowRISC SystemVerilog Coding Style Guide for Design Verification
+
+This document is a supplemental Style Guide to the [lowRISC Verilog style
+guide](https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md),
+with emphasis on writing code for Design Verification (DV) in SystemVerilog,
+following the UVM methodology.
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Naming and Style](#naming-and-style)
+* [Directory Structure and Packages](#directory-structure-and-packages)
+* [UVM Guidelines](#uvm-guidelines)
+  * [Class Definitions](#class-definitions)
+  * [UVM new functions](#uvm-new-functions)
+  * [uvm_scoreboard usage](#uvm_scoreboard-usage)
+  * [uvm_agent usage](#uvm_agent-usage)
+  * [uvm_driver usage](#uvm_driver-usage)
+  * [Macro Usage](#macro-usage)
+  * [Factory](#factory)
+  * [Configuration Mechanism](#configuration-mechanism)
+  * [Sequences](#sequences)
+  * [Sequence Items](#sequence-items)
+  * [RAL Usage](#ral-usage)
+  * [Objections and Coordinating End of Test](#objections-and-coordinating-end-of-test)
+  * [Logging and Print Messages](#logging-and-print-messages)
+  * [DPI and C Connections](#dpi-and-c-connections)
+* [SystemVerilog Language Features](#systemverilog-language-features)
+  * [Randomization](#randomization)
+  * [Enums](#enums)
+  * [Interfaces, Clocking Blocks, Modports](#interfaces,-clocking-blocks,-modports)
+  * [Loop Operators](#loop-operators)
+  * [Code Within Asserts](#code-within-asserts)
+  * [Wait and Fork](#wait-and-fork)
+  * [Void Casts](#void-casts)
+  * [Associative Arrays](#associative-arrays)
+  * [Bind Statements](#bind-statements)
+  * [Simulator Specific Code](#simulator-specific-code)
+  * [Forbidden System Tasks and Functions](#forbidden-system-tasks-and-functions)
+
+
+
+## Introduction
+
+This guide defines the Comportable style for DV-specific SystemVerilog code. The
+goal is to use this common set of guidelines when implementing testbenches using
+the UVM methodology to allow all developed Comportable DV code to be uniform,
+reusable, and portable.
+This document assumes previous experience and working knowledge of the UVM, and
+as such all included code snippets will be fairly concise.
+A testbench generation script
+[uvmdvgen.py](https://github.com/lowRISC/opentitan/blob/master/util/uvmdvgen.py)
```

Comment:
@sriyerg and I had a brief discussion regarding this, and this script will likely follow the dv base libraries into the separate lowRISC IP repository. For the time being, I refer to this script in the style guide just to provide another option for testbench generation, especially because much of the testbench style guidelines in the document are mirrored in the auto-generated UVM files produced by this script.

Reply:
SGTM, thanks